pkg/kube/secret/manager.go

```diff
@@ -22,31 +23,30 @@ func NewSecretManager(clientset kubernetes.Interface) *Manager {
 }
 
 // GetImagesWithCredentials gets private images for the specified PodSpec and maps them to the Docker ServerCredentials.
-func (s *Manager) GetImagesWithCredentials(ctx context.Context, namespace string, spec core.PodSpec) (credentials map[string]docker.ServerCredentials, err error) {
+func (s *Manager) GetImagesWithCredentials(ctx context.Context, namespace string, spec corev1.PodSpec) (credentials map[string]docker.Auth, err error) {
 	images := s.GetImages(spec)
 
-	serverCredentials, err := s.GetServersWithCredentials(ctx, namespace, spec.ImagePullSecrets)
+	serverCredentials, err := s.GetRegistryHostsWithCredentials(ctx, namespace, spec.ImagePullSecrets)
 	if err != nil {
 		return
 	}
 
-	credentials = make(map[string]docker.ServerCredentials)
+	credentials = make(map[string]docker.Auth)
 	for _, image := range images {
-		server := s.GetServerFromImage(image)
+		server, err := docker.GetServerFromImageRef(image)
```

Comment:
Here was the actual bug where imageRef for private repository was not parsed properly. For example, we returned `some-namespace` for `some-namespace/my-image:my-tag` instead of returning `index.docker.io`.